jax/_src/lax/slicing.py

```diff
@@ -686,10 +684,10 @@ def index_in_dim(operand: Array, index: int, axis: int = 0,
     return lax.squeeze(result, (axis,))
 
 
-def dynamic_slice_in_dim(operand: Array, start_index: Array,
+def dynamic_slice_in_dim(operand: Array, start_index: ArrayLike,
                          slice_size: int, axis: int = 0) -> Array:
   """Convenience wrapper around dynamic_slice applying to one dimension."""
-  start_indices = [np.zeros((), dtype=dtypes.dtype(start_index))] * operand.ndim
+  start_indices = [lax._zero(start_index)] * operand.ndim
```

Comment:
Any reason you need to change this back? I remember we added it in https://github.com/google/jax/pull/12566 to avoid some unenecessary staging.

Reply:
The problem is that `np.zeros` is of type `np.ndarray`, so this was leading to a typecheck error when an `Array` was assigned to the list below. I could figure out how to annotate this differently to keep the original if you think that would be better.

Reply:
I think that would be nice but interested in hearing from @mattjj

Reply:
Switched it back to a constant; also did the same in `dynamic_update_slice`